mono/utils/mono-state.c

```diff
@@ -927,6 +927,75 @@ mono_native_state_add_memory (MonoStateWriter *writer)
 	mono_state_writer_printf(writer, "},\n");
 }
 
+#define MONO_CRASH_REPORTING_MAPPING_LINE_LIMIT 30
+
+static void
+mono_native_state_add_process_map (MonoStateWriter *writer)
+{
+#if defined(__linux__) && !defined(HOST_ANDROID)
+	static gint32 handle_in_use;
+	gint32 old = mono_atomic_cas_i32 (&handle_in_use, TRUE /* set */, FALSE /* compare */);
+	if (old != FALSE) {
+		g_async_safe_printf ("Warning: Couldn't open /proc/self/maps for concurrent reads, crash report will lack it.\n");
+		return;
+	}
+
+	mono_memory_barrier ();
+
+	int handle = g_open ("/proc/self/maps", O_RDONLY, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+	if (handle == -1) {
+		g_async_safe_printf ("Couldn't find /proc/self/maps on Linux system. Continuing.");
+		return;
+	}
+
+	assert_has_space (writer);
+	mono_state_writer_indent (writer);
+	mono_state_writer_object_key (writer, "process_map");
+	mono_state_writer_printf(writer, "[\n");
+
+	int mapping = 0;
+	while (mapping < MONO_CRASH_REPORTING_MAPPING_LINE_LIMIT) {
+		if (mapping > 0)
+			mono_state_writer_printf (writer, "\",\n");
+
+		mono_state_writer_printf (writer, "\t\"");
```

Comment:
Can this occur at end of file w/o anything in the quotes?

Reply:
Try to raise the line limit to fit it all?

Reply:
The line limit is a limit on the size of the produced JSON. It was present in the initial dumper because I assumed it was possible to find apps that had an unreasonably large number of mappings. 

If this occurs at the end of the file, won't it just be an empty array element at the end? That seems like a fine edge case. 

[ "foo", "" ]

Reply:
Fair enough.